cli/medperf/tests/comms/entity_resources/sources/test_synapse.py

```diff
@@ -0,0 +1,103 @@
+from medperf.comms.entity_resources.sources.synapse import SynapseSource
+import pytest
+import os
+from medperf.exceptions import (
+    CommunicationAuthenticationError,
+    CommunicationRetrievalError,
+)
+from synapseclient.core.exceptions import SynapseError
+import synapseclient
+
+
+class MockSynapseFile:
+    def __init__(self, name):
+        self.name = name
+
+
+@pytest.fixture
+def synapse_client(mocker):
+    mock = mocker.create_autospec(synapseclient.Synapse)
+    mocker.patch.object(synapseclient, "Synapse", return_value=mock)
+
+
+def test_download_works_as_expected(mocker, fs, synapse_client):
+    # Arrange
+    outpath = "path/to/out.tar.gz"
+    contents = "contents"
+    syn_id = "syn535353"
+    synapse_source = SynapseSource()
```

Comment:
This arrange block above seems to be repeated on multiple tests. It might be best to put it inside a setup fixture for those tests